.github/workflows/clang-tidy.yml

```
@@ -0,0 +1,61 @@
+name: Clang-Tidy
+
+on:
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+env:
+  BUILD_TYPE: Release
+
+jobs:
+  check-needed:
+    runs-on: ubuntu-latest
+
+    outputs:
+      status: ${{ steps.needed.outcome }}
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Checkout base ref
+      id: base-ref
+      run: |
+        if [ $GITHUB_BASE_REF ]; then 
+          git fetch origin "${GITHUB_BASE_REF}:refs/remotes/origin/${GITHUB_BASE_REF}"
+          echo "base=origin/${GITHUB_BASE_REF}" >> "$GITHUB_OUTPUT"
+        else
+          git fetch origin ${{ github.event.before }}
+          echo "base=${{ github.event.before }}" >> "$GITHUB_OUTPUT"
+        fi
+    - name: Check if clang-tidy required
+      id: needed
+      continue-on-error: true
+      run: git diff --name-only -U0 ${{ steps.base-ref.outputs.base }} | grep -c -e "\.cpp$" -e "\.h\.in$" -e "\.h\.in\.cmake$" -e "\.h$" -e "\.hpp"
+    
+  run-tidy:
+    needs: check-needed
+    if: needs.check-needed.outputs.status == 'success'
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+    - name: Checkout base ref
+      id: base-ref
+      run: |
+        if [ $GITHUB_BASE_REF ]; then 
+          git fetch origin "${GITHUB_BASE_REF}:refs/remotes/origin/${GITHUB_BASE_REF}"
+          echo "base=origin/${GITHUB_BASE_REF}" >> "$GITHUB_OUTPUT"
+        else
+          git fetch origin ${{ github.event.before }}
+          echo "base=${{ github.event.before }}" >> "$GITHUB_OUTPUT"
+        fi
+    - name: Configure CMake
+      run: cmake -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_EXPORT_COMPILE_COMMANDS=ON -DBUILD_TESTS=ON -DBUILD_LENSTOOL=ON
+    - name: Install clang-tidy for clang-tidy-diff
+      run: sudo apt update && sudo apt install clang-tidy
+    - name: Run clang-tidy
+      run: git diff -U0 ${{ steps.base-ref.outputs.base }} | clang-tidy-diff -use-color -p1 -regex .*cpp -path ${{github.workspace}}/build > out 2> err
+    - name: Check results
+      run: if grep -q "warnings treated as errors" err; then cat out; false; else true; fi
```

```suggestion
      run: if grep -q "warnings\? treated as errors" err; then cat out; false; else true; fi
```
In case there is only one warning